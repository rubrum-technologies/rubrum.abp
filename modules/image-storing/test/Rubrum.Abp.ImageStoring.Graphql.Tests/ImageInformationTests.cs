using CookieCrumble;
using HotChocolate.Types.Relay;
using Shouldly;
using Xunit;
using static Rubrum.Abp.ImageStoring.ImageStoringTestConstants;

namespace Rubrum.Abp.ImageStoring;

public sealed class ImageInformationTests : ImageStoringGraphqlTestBase
{
    private readonly IImageInformationRepository _repository;
    private readonly IIdSerializer _idSerializer;

    public ImageInformationTests()
    {
        _repository = GetRequiredService<IImageInformationRepository>();
        _idSerializer = GetRequiredService<IIdSerializer>();
    }

    [Fact]
    public async Task FetchUrlById()
    {
        var image = await _repository.GetAsync(JpegId);
        var id = _idSerializer.Serialize(null, ImageInformationConstants.TypeName, image.Id);

        await using var result = await ExecuteRequestAsync(b => b.SetQuery(
            $$"""
              query {
                  imageUrl(id: "{{id}}")
              }
              """));

        result.ShouldNotBeNull();

        result.MatchSnapshot();
    }

    [Fact]
    public async Task FetchUrlsByTag()
    {
        await using var result = await ExecuteRequestAsync(b => b.SetQuery(
            $$"""
              query {
                  imageUrlsByTag(tag: "{{ImageTag}}")
              }
              """));

        result.ShouldNotBeNull();

        result.MatchSnapshot();
    }

    [Fact]
    public async Task FetchById()
    {
        var id = _idSerializer.Serialize(null, ImageInformationConstants.TypeName, GifId);

        await using var result = await ExecuteRequestAsync(b => b.SetQuery(
            $$"""
              query {
                  image(id: "{{id}}") {
                    id
                    tag
                    fileName
                    systemFileName
                    isDisposable
                    entityVersion
                    url
                  }
              }
              """));

        result.ShouldNotBeNull();

        result.MatchSnapshot();
    }

    [Fact]
    public async Task FetchByTag()
    {
        await using var result = await ExecuteRequestAsync(b => b.SetQuery(
            $$"""
              query {
                  imagesByTag(tag: "{{ImageTag}}"){
                    id
                    tag
                    fileName
                    systemFileName
                    isDisposable
                    entityVersion
                    url
                  }
              }
              """));

        result.ShouldNotBeNull();

        result.MatchSnapshot();
    }
}
