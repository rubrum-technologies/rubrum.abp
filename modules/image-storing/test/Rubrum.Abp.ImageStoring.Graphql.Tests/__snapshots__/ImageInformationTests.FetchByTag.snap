{
  "data": {
    "imagesByTag": [
      {
        "id": "SW1hZ2VJbmZvcm1hdGlvbgpnZjM2NGExZWMzOWEyNGRlZTgwYWE2YjllYjQyMmFlYmI=",
        "tag": "98ffe60d-6096-4504-8afe-2f963e32a961",
        "fileName": "1.svg",
        "systemFileName": "f364a1ec-39a2-4dee-80aa-6b9eb422aebb.webp",
        "isDisposable": false,
        "entityVersion": 0,
        "url": "/api/image-storing/images/download/f364a1ec-39a2-4dee-80aa-6b9eb422aebb"
      },
      {
        "id": "SW1hZ2VJbmZvcm1hdGlvbgpnY2IwZGI1MjliYjY1NDIyNmFmMGVhYzM5NjE0YmI0N2Y=",
        "tag": "98ffe60d-6096-4504-8afe-2f963e32a961",
        "fileName": "3.jpg",
        "systemFileName": "cb0db529-bb65-4226-af0e-ac39614bb47f.webp",
        "isDisposable": false,
        "entityVersion": 0,
        "url": "/api/image-storing/images/download/cb0db529-bb65-4226-af0e-ac39614bb47f"
      },
      {
        "id": "SW1hZ2VJbmZvcm1hdGlvbgpnNDhmZmJiM2I3NWRlNDM0NjkxYzlkMjRhZTA2YWQ2ODI=",
        "tag": "98ffe60d-6096-4504-8afe-2f963e32a961",
        "fileName": "5.png",
        "systemFileName": "48ffbb3b-75de-4346-91c9-d24ae06ad682.webp",
        "isDisposable": false,
        "entityVersion": 0,
        "url": "/api/image-storing/images/download/48ffbb3b-75de-4346-91c9-d24ae06ad682"
      }
    ]
  }
}